test-helpers/src/shotover_process.rs

```diff
@@ -67,6 +69,12 @@ impl ShotoverProcessBuilder {
         self
     }
 
+    pub fn with_args(mut self, args: &[&str]) -> Self {
```

Comment:
Rather than have each test define the args they provide, `ShotoverProcessBuilder` provides methods for each argument that we want to configure in tests.
So in this case, lets add a method `fn with_hotreload(enabled: bool)` instead of allowing tests to set arbitrary args.